### DISCUSSION TITLE
Warning in neomodel documentation

### DISCUSSION THREAD

**Discussion by DaraSilveraM** [neo4j-contrib/neomodel#754] [General]
Hi there! I'm going to talk about neomodel in the Us DjangoCon, and I have a question regarding this warning, because i could not understand it quite well:

```
Warning
This feature is still a work in progress for extending path traversal and fecthing. It currently stops at returning the resolved objects as they are returned in Cypher. So for instance, if your path looks like (startNode:Person)-[r1]->(middleNode:City)<-[r2]-(endNode:Country), then you will get a list of results, where each result is a list of (startNode, r1, middleNode, r2, endNode). These will be resolved by neomodel, so startNode will be a Person class as defined in neomodel for example
```

apart from introducing neomodel to the attendees I want to mention some negative aspects too, or things to bare in mind. This talk is going to be on October 16th, so I hope I receive answers before that to enrich the presentation!

**Comment by aanastasiou** [comment]
Hello @DaraSilvera-Oct , this is a bit late, I know, but here it goes:

* The feature you mention is indeed work in progress, but the situation is better from 5.1.1 onwards with the Neomodel implementation of a `Path` object. So the wording in that warning is likely to change soon.

* A negative aspect (also soon to be corrected) is that the way Neomodel currently works, it cannot deal easily with multiple inheritance. Very briefly, if your data model looks like a tree, where there is a straight lineage from a given child class to its parent, Neomodel would be able to deal with it. If you have crossing paths (e.g. the [Diamond problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem)) then _at the moment_, Neomodel would fail to instantiate the appropriate class in your data model.

Other than this, I am not aware of any other real "show stoppers" to its use.
